cmd/manager/main.go

```diff
@@ -136,11 +143,18 @@ func main() {
 		klog.Fatal(err)
 	}
 
+	configManagedClient, startCache, err := newConfigManagedClient(mgr.GetConfig())
+	if err != nil {
+		klog.Fatal(err)
+	}
+	mgr.Add(startCache)
+
 	// Initialize machine actuator.
 	machineActuator := machineactuator.NewActuator(machineactuator.ActuatorParams{
-		Client:           mgr.GetClient(),
-		EventRecorder:    mgr.GetEventRecorderFor("awscontroller"),
-		AwsClientBuilder: awsclient.NewValidatedClient,
+		Client:              mgr.GetClient(),
+		EventRecorder:       mgr.GetEventRecorderFor("awscontroller"),
+		AwsClientBuilder:    awsclient.NewValidatedClient,
+		ConfigManagedClient: configManagedClient,
```

Comment:
you could also just use `mgr.GetAPIReader`, not sure it satisfies your needs.

Reply:
I had a look through the code that sets up the API Reader and it seems it is unaffected by the namespace flag, as such should be usable, only downside being that it doesn't have caching, how much of an issue is that? I'm not entirely sure

Reply:
Yes, it was the caching that I wanted to get. In the [PR](https://github.com/openshift/cluster-api-provider-aws/pull/372#discussion_r534495354) that introduced the fetch of the kube-cloud-config ConfigMap, @michaelgugino had concerns about querying the API server on every reconcile. I think those concerns are justified.